app/screens/global_threads/threads_list/threads_list.tsx

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useMemo, useState, useRef} from 'react';
+import {useIntl} from 'react-intl';
+import {FlatList, Platform, StyleSheet} from 'react-native';
+
+import {fetchThreads} from '@actions/remote/thread';
+import {General} from '@app/constants';
+import Loading from '@components/loading';
+
+import EmptyState from './empty_state';
+import EndOfList from './end_of_list';
+import Header, {Tab} from './header';
+import Thread from './thread';
+
+import type ThreadModel from '@typings/database/models/servers/thread';
+
+export type {Tab};
+
+export type Props = {
+    setTab: (tab: Tab) => void;
+    serverUrl: string;
+    tab: Tab;
+    teamId: string;
+    teammateNameDisplay: string;
+    testID: string;
+    threads: ThreadModel[];
+    theme: Theme;
+    unreadsCount: number;
+};
+
+const styles = StyleSheet.create({
+    messagesContainer: {
+        flexGrow: 1,
+    },
+    loadingStyle: {
+        alignItems: 'center',
+        flex: 1,
+        justifyContent: 'center',
+        paddingTop: 16,
+        paddingBottom: 8,
+    },
+});
+
+const ThreadsList = ({
+    setTab,
+    tab,
+    serverUrl,
+    teamId,
+    teammateNameDisplay,
+    testID,
+    theme,
+    threads,
+    unreadsCount,
+}: Props) => {
+    const intl = useIntl();
+    const hasCalled = useRef(false);
+    const [isLoading, setIsLoading] = useState(true);
+    const [endReached, setEndReached] = useState(false);
+
+    const noThreads = !threads?.length;
+    const lastThread = threads?.length ? threads[threads.length - 1] : null;
+
+    useEffect(() => {
+        // this is to be called only when there are no threads
```

Comment:
If it should only be called when there are no threads, won't be better to keep the dependencies only to `noThreads`? Or there are reason to execute this snippet also when other of the dependencies change?